prelude/bootstrap.js

```diff
@@ -25,6 +25,60 @@ const { promisify, _extend } = require('util');
 const { Script } = require('vm');
 const { tmpdir } = require('os');
 
+/**
+ * note: developers may activate the doDebug flag to enable a trace in the console
+ *       at runtime.
+ *       this is particularly useful for the maintenance of the bootstrap.js file
+ *       as pkg doesn't allow debugging the packed code.
+ */
+const doDebug = false;
+
+let debugIndent = 1;
+const debugLog = doDebug
+  ? function cc() {
+      const a = [' '.padEnd(debugIndent * 2), ...arguments];
+      // eslint-disable-next-line no-console
+      console.log.apply(console, a);
+    }
+  : () => {};
+
+function debugWrap(f) {
+  if (!doDebug) {
+    return f;
+  }
+  return function a() {
+    // eslint-disable-next-line no-console
+    debugLog('entering function ', f.name);
+    debugIndent += 1;
+
+    const lastArgument = arguments[arguments.length - 1];
+    if (typeof lastArgument === 'function') {
+      arguments[arguments.length - 1] = function myCallback() {
+        debugIndent -= 1;
+        lastArgument.apply(null, arguments);
+      };
+      f.apply(this, arguments);
+    } else {
+      const ret = f.apply(this, arguments);
+      debugIndent -= 1;
+      return ret;
+    }
+  };
+}
```

Comment:
Instead of a debugWrap function that could mess up the function calling context I would suggest to create a function that receives the function context as parameter.

```
function debugCall(context, args) {...}

// inside functions:
debugCall(this, arguments)
```

Reply:
Can you be more specific?

Reply:
Now it is:

debugWrap(function f() {})

I suggest:

function f() {
 debugCall(this, arguments)
}

Reply:
OK , I see.

I don't think that this would work. 

We need to capture the name of the wrapped function.

Also we want to have no effect on the code ( same behavior as before ) when debug === false , in which case debugWrap is a passthrough

Reply:
Ok was just an idea :)